packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_MappingTransformer.ts

```diff
@@ -463,7 +467,11 @@ const transformRelationalPropertyMapping = (
           [SOURCE_INFORMATION_KEY],
         ),
   ) as V1_RawRelationalOperationElement;
-  propertyMapping.source = undefined; // @MARKER: GRAMMAR ROUNDTRIP --- omit this information during protocol transformation as it can be interpreted while building the graph
+  // Note: renderSourceId is required to have the source information persisted in case of relationalPropertyMappings
+  // source key needs to be interpeted here for consistency between engine protcol and studio protcol
+  propertyMapping.source = renderSourceId
+    ? transformPropertyMappingSource(element.sourceSetImplementation)
```

Comment:
should this be for all relational property mappings or only within association set implementations?